app/uk/gov/hmrc/cgtpropertydisposals/controllers/SubscriptionController.scala

```diff
@@ -72,14 +88,34 @@ class SubscriptionController @Inject()(
         case BackendError(e) =>
           logger.warn("Error while trying to subscribe", e)
           InternalServerError
-        case EnrolmentError(cgtReference) =>
-          logger.warn("Error while trying to allocate enrolment")
-          Ok(Json.toJson(SubscriptionResponse(cgtReference))) // We return a 200 OK even though it is an error as we will retry asynchronously
+        case FailedTaxEnrolmentCall(taxEnrolmentRequest, cgt) =>
+          logger.warn("Error while trying to allocate enrolment - sending message to retry asynchronously")
+          retry(taxEnrolmentRequest).value.map {
+            case Left(error) =>
+              logger.warn(s"Failed to submit async retry request for $taxEnrolmentRequest: error [$error]")
```

Comment:
it feels like we should send an error response back to the request if we can't put the retry onto the queue - otherwise the enrolment will never get retried again. What will happen is the user will see that the subscription confirmation screen and then the next time they log in they'd be asked to subscribe again since we'd have no enrolment of them and we wouldn't be able to tell they've failed an enrolment. This'll be a confusing journey for the user.
 
If we fail the request, as far as the user is concerned they're not subscribed and will know to try again. When the user tries to subscribe again, then it'll trigger another attempt to enrol and if it fails we'll try to put it on the queue again.